myresources.py

```diff
@@ -309,24 +309,36 @@ def usage_string(job):
             job[res]['usage'],
             usage_for_free=usage_for_free,
             empty_bar=empty_bar,
-            nocol=nocolor,
+            nocol=NOCOLOR,
             show_rating=show_rating,
-            lev=levels[res],
+            lev=LEVELS[res],
         )
 
         res_fullstrings[res] = ' '.join([
-            res_names[res].rjust(12),
+            RES_NAMES[res].rjust(12),
             a_ustr['used'].rjust(10),
-            units[res].rjust(2),
+            UNITS[res].rjust(2),
             a_ustr['avail'].rjust(10),
-            units[res].rjust(2),
+            UNITS[res].rjust(2),
             usagestr.rjust(6),
             ubar.rjust(31),
             res_extrastrings[res],
         ])
 
-    return '\n'.join(res_fullstrings[res] for res in reslist)
+    return '\n'.join(res_fullstrings[res] for res in RESLIST)
 
+def csv_string(job):
+    fulllist = [
+        job['jobid'],
+        job['state'],
+        job['jobname'],
+    ]
+    for res in RESLIST:
+        fulllist.extend([
+            '%s' % job[res]['avail'] if job[res]['avail'] is not None else '',
+            '%s' % job[res]['used'] if job[res]['used'] is not None else '',
+        ])
+    return ','.join(fulllist)
```

Comment:
use the `csv` module, it will handle all corner cases for you

Reply:
good idea, done